pydocteur/github_api.py

```diff
@@ -59,3 +60,42 @@ def has_pr_number(payload) -> bool:
         return False
     else:
         return True
+
+
+def get_coauthors(pr) -> set:
+    co_authored = set()
+    compiled_regex = re.compile("(Co-authored-by:.*)")
+    for commit in pr.get_commits():
+        commit_message = commit.commit.message
+        matches = compiled_regex.findall(commit_message)
+        if matches:
+            for item in matches:
+                co_authored.add(item)
+    return co_authored
+
+
+def get_issues_to_close(body):
+    if not body or body == "":
+        return []
+    results = re.findall("(?:close[sd]?|fix|fixe[sd]|resolve[sd]?)\\s+(#\\d+)", body, flags=re.IGNORECASE)
+    return results
+
+
+def get_title_message_for_merge(pr):
+    logger.info(f"Generating title and message for merge of PR #{pr.number}")
+    co_authors = get_coauthors(pr)
+    closing_issues = get_issues_to_close(pr.body)
+
+    fixes = "Closes ".join(closing_issues)
+    coauthors = "\n".join(co_authors)
+
+    title = "Automerge of PR #{pr_number} by @{author}".format(pr_number=pr.number, author=pr.user.login)
```

Comment:
If I understand it correctly, the "title" is the first line of the commit. If we don't use the PR title, but set our title, the `git log` will looks strange, like:
```
* d4b9a0d (HEAD -> sitem
* 76cdfb2 Automerge of PR 8 by @...
* c23b401 Automerge of PR 7 by @...
* fa10a70 Automerge of PR 6 by @...
* bde3ad0 Automerge of PR 5 by @...
* 815baf7 Automerge of PR 4 by @...
```

I don't like it ;p

Reply:
Agreed. Will be changed.